BUG: failure in `scipy/optimize/tests/test_minimize_constrained.py::test_gh11649`

### Describe your issue.

There is a test failure in main causing most CI jobs to fail, e.g. see gh-21744.

### Reproducing Code Example

```python
-
```


### Error message

```shell
================================== FAILURES ===================================
________________________________ test_gh11649 _________________________________

    def test_gh11649():
        bnds = Bounds(lb=[-1, -1], ub=[1, 1], keep_feasible=True)
    
        def assert_inbounds(x):
            assert np.all(x >= bnds.lb)
            assert np.all(x <= bnds.ub)
    
        def obj(x):
            assert_inbounds(x)
            return np.exp(x[0])*(4*x[0]**2 + 2*x[1]**2 + 4*x[0]*x[1] + 2*x[1] + 1)
    
        def nce(x):
            assert_inbounds(x)
            return x[0]**2 + x[1]
    
        def nce_jac(x):
            return np.array([2*x[0], 1])
    
        def nci(x):
            assert_inbounds(x)
            return x[0]*x[1]
    
        x0 = np.array((0.99, -0.99))
        nlcs = [NonlinearConstraint(nci, -10, np.inf),
                NonlinearConstraint(nce, 1, 1, jac=nce_jac)]
    
        res = minimize(fun=obj, x0=x0, method='trust-constr',
                       bounds=bnds, constraints=nlcs)
>       assert res.success
E       AssertionError: assert False
E        +  where False =            message: Constraint violation exceeds 'gtol'\n           success: False\n            status: 4\n              ...enalty: 1.0\n barrier_parameter: 2.048000000000001e-09\n barrier_tolerance: 2.048000000000001e-09\n             niter: 225.success

C:\hostedtoolcache\windows\Python\3.11.9\x64\Lib\site-packages\scipy\optimize\tests\test_minimize_constrained.py:741: AssertionError
```


### SciPy/NumPy/Python version and system information

```shell
main
```

--- a/scipy/optimize/_trustregion_constr/minimize_trustregion_constr.py
+++ b/scipy/optimize/_trustregion_constr/minimize_trustregion_constr.py
@@ -302,6 +302,10 @@ def _minimize_trustregion_constr(fun, x0, args, grad,
         * 3 : `callback` function requested termination.
         * 4 : Constraint violation exceeds 'gtol'.
 
+        .. versionchanged:: 1.15.0
+            If the constraint violation exceeds gtol then `result.success`
+            will be False. Previously this could've been reported as True.
+
     cg_stop_cond : int
         Reason for CG subproblem termination at the last iteration:
 

--- a/scipy/optimize/_trustregion_constr/minimize_trustregion_constr.py
+++ b/scipy/optimize/_trustregion_constr/minimize_trustregion_constr.py
@@ -303,7 +303,7 @@ def _minimize_trustregion_constr(fun, x0, args, grad,
         * 4 : Constraint violation exceeds 'gtol'.
 
         .. versionchanged:: 1.15.0
-            If the constraint violation exceeds gtol, then `result.success`
+            If the constraint violation exceeds `gtol`, then ``result.success``
             will now be False.
 
     cg_stop_cond : int

--- a/scipy/optimize/_trustregion_constr/minimize_trustregion_constr.py
+++ b/scipy/optimize/_trustregion_constr/minimize_trustregion_constr.py
@@ -303,8 +303,8 @@ def _minimize_trustregion_constr(fun, x0, args, grad,
         * 4 : Constraint violation exceeds 'gtol'.
 
         .. versionchanged:: 1.15.0
-            If the constraint violation exceeds gtol then `result.success`
-            will be False. Previously this could've been reported as True.
+            If the constraint violation exceeds gtol, then `result.success`
+            will now be False.
 
     cg_stop_cond : int
         Reason for CG subproblem termination at the last iteration:

--- a/scipy/optimize/tests/test_minimize_constrained.py
+++ b/scipy/optimize/tests/test_minimize_constrained.py
@@ -711,6 +711,9 @@ def test_bug_11886():
 
 
 def test_gh11649():
+    # trust - constr error when attempting to keep bound constrained solutions
+    # feasible. Algorithm attempts to go outside bounds when evaluating finite
+    # differences. (don't give objective an analytic gradient)
     bnds = Bounds(lb=[-1, -1], ub=[1, 1], keep_feasible=True)
 
     def assert_inbounds(x):
@@ -721,20 +724,20 @@ def test_gh11649():
         assert_inbounds(x)
         return np.exp(x[0])*(4*x[0]**2 + 2*x[1]**2 + 4*x[0]*x[1] + 2*x[1] + 1)
 
-    def nce(x):
-        assert_inbounds(x)
-        return x[0]**2 + x[1]
-
-    def nce_jac(x):
-        return np.array([2*x[0], 1])
+    # def nce(x):
+    #     assert_inbounds(x)
+    #     return x[0]**2 + x[1]
+    #
+    # def nce_jac(x):
+    #     return np.array([2*x[0], 1])
 
     def nci(x):
         assert_inbounds(x)
         return x[0]*x[1]
 
     x0 = np.array((0.99, -0.99))
-    nlcs = [NonlinearConstraint(nci, -10, np.inf),
-            NonlinearConstraint(nce, 1, 1, jac=nce_jac)]
+    nlcs = [NonlinearConstraint(nci, -10, np.inf),]
+            # NonlinearConstraint(nce, 1, 1, jac=nce_jac)]
 
     res = minimize(fun=obj, x0=x0, method='trust-constr',
                    bounds=bnds, constraints=nlcs)

--- a/scipy/optimize/tests/test_minimize_constrained.py
+++ b/scipy/optimize/tests/test_minimize_constrained.py
@@ -724,24 +724,23 @@ def test_gh11649():
         assert_inbounds(x)
         return np.exp(x[0])*(4*x[0]**2 + 2*x[1]**2 + 4*x[0]*x[1] + 2*x[1] + 1)
 
-    # def nce(x):
-    #     assert_inbounds(x)
-    #     return x[0]**2 + x[1]
-    #
-    # def nce_jac(x):
-    #     return np.array([2*x[0], 1])
+    def nce(x):
+        assert_inbounds(x)
+        return x[0]**2 + x[1]
+
+    def nce_jac(x):
+        return np.array([2*x[0], 1])
 
     def nci(x):
         assert_inbounds(x)
         return x[0]*x[1]
 
     x0 = np.array((0.99, -0.99))
-    nlcs = [NonlinearConstraint(nci, -10, np.inf),]
-            # NonlinearConstraint(nce, 1, 1, jac=nce_jac)]
+    nlcs = [NonlinearConstraint(nci, -10, np.inf),
+            NonlinearConstraint(nce, 1, 1, jac=nce_jac)]
 
     res = minimize(fun=obj, x0=x0, method='trust-constr',
                    bounds=bnds, constraints=nlcs)
-    assert res.success
     assert_inbounds(res.x)
     assert nlcs[0].lb < nlcs[0].fun(res.x) < nlcs[0].ub
 

